File: daoduytinhjavaweb/src/main/java/com/daoduytinh/service/SalientService.java
package com.daoduytinh.service;

import java.util.List;
import java.util.Map;

import javax.validation.Valid;

import org.springframework.data.domain.Pageable;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Service;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestBody;
import com.daoduytinh.entity.Salient;
import com.daoduytinh.exception.ResourceNotFoundException;
@Service
public interface SalientService {

	public Iterable<Salient> getAllSalient();
	
    public ResponseEntity<Salient> getSalientById(@PathVariable(value = "id") Long Id)
        throws ResourceNotFoundException;
	
    List<Salient> getSalientByProductId(@PathVariable(value = "id") Long Id)
            throws ResourceNotFoundException ;
            
    public Salient createSalient(@Valid @RequestBody Salient salient,@PathVariable(value = "id") Long Id);
	
    public ResponseEntity<Salient> updateSalient(@PathVariable(value = "id") Long Id,@PathVariable(value = "idpro") Long Idpro,
         @Valid @RequestBody Salient salientDetail) throws ResourceNotFoundException ;

    public Map<String, Boolean> deleteSalient(@PathVariable(value = "id") Long Id)
         throws ResourceNotFoundException;
}

File: ck-editor5-image-upload-master/Client/src/service/SalientService.js
import axios from 'axios';

const SALIENT_REST_API ="http://localhost:8080/salient";

class SalientService{

    getAllNews(){
        return axios.get(SALIENT_REST_API);
    }
    getSalientById(id){
        return axios.get(SALIENT_REST_API+'/'+id);
    }
    getSalientbyProduct(id){
        return axios.get("http://localhost:8080/salientbypro/"+id);
    }
    deleteSalient(id){
        return axios.delete(SALIENT_REST_API+'/'+id);
    }
    updateSalient(salient,id,idpro){
        return axios.put(SALIENT_REST_API+'/'+id+'/'+idpro,salient);
    }
    createSalient(salient,id){
        return axios.post(SALIENT_REST_API+"/"+id,salient);
    }
}
export default new SalientService();